src/components/index.js

```diff
@@ -148,6 +148,10 @@ class Joyride extends React.Component {
     // Update the index if the first step is not found
     if (!controlled && status === STATUS.RUNNING && index === 0 && !target) {
       this.store.update({ index: index + 1 });
+    }
```

Comment:
This will break the current functionality

Reply:
Any suggestions to solve this issue?

Reply:
Do you really need a controlled tour using the `stepIndex`?
Because if you do, you should handle these problems by yourself and that's why there's a `!controlled` in the conditional.

Reply:
I was just playing. with the code to trigger the missing target callback, but it will be triggered multiple times because the internal index doesn't change.

I'll try to dig deeper later

Reply:
thanks @gilbarbara 
needed this for implementing joyride in different routes, since the routes might take some time to get loaded, I will use the  error:target_not_found event to pause and start the tour in 1000ms

Reply:
I think this approach could have nasty side effects.

You can use a mutation observer to handle that instead.
Or a centralized state with React.Context or similar.

Reply:
The problem is that the components inside the route are lazy loaded, in that case it will be difficult I guess.

Reply:
If the `Context.Provider` is wrapping the content, it doesn't matter.
The Consumer can update the state even if lazy-loaded.